src/modules/authentication/components/Registration/Registration.tsx

```diff
@@ -1,5 +1,231 @@
-function Registration() {
-	return <div></div>;
+import styles from './register.module.css'
+import logo from '../../../../assets/images/PMS 3.svg'
+import profileImg from '../../../../assets/images/profile.svg'
+import {useForm ,SubmitHandler }  from 'react-hook-form'
+import { toast } from "react-toastify";
+import { useState , useEffect} from 'react';
+import { useNavigate } from 'react-router-dom';
+import { AUTH_URLS, axiosInstance, BASE_AUTH} from '../../../../api';
+
+
+export default function Registration() {
+
+	const [isVisible, setVisible] = useState<boolean>(false);  
+	const [isRePasswordVisible, setIsRePasswordVisible] = useState<boolean>(false);  
+	const [isPasswordShown, setIsPasswordShown] = useState<boolean>(false)
+	const [isRePasswordShown, setIsRePasswordShown] = useState<boolean>(false)
+
+	const toggleHideShowPassword = ():void => {
+		setVisible(!isVisible);
+		
+		setIsPasswordShown(!isPasswordShown)
+		
+	  };
+	const toggleHideShowRePassword = ():void => {
+		setIsRePasswordVisible(!isRePasswordVisible);
+		
+		setIsRePasswordShown(!isRePasswordShown)
+		
+	  };
+
+
+const navigate = useNavigate()
+type FormFields = {
+	userName : string , 
+	email: string,
+	phoneNumber : number ,
+	password: number | string,
+	confirmPassword : number | string,
+	country : string , 
+	profileImage : string | File | HTMLImageElement
 }
 
-export default Registration;
+const{ 
+	formState:{errors,isSubmitting },
+	register  ,
+	handleSubmit,watch
+}=	useForm <FormFields>({ mode:'onChange'})
+
+useEffect(()=>{
+	if(watch('confirmPassword')) {
+	  trigger('confirmPassword')
+
+	}
+  },[watch('password')])
+
+
+const onSubmit : SubmitHandler<FormFields> = async (data)=> {
+
+	await axiosInstance.post(BASE_AUTH+AUTH_URLS.register , data).then((resp)=>{
+		console.log(resp)
+		toast.success(resp?.data?.message || 'account created successfully')
+		navigate('/verify-user')
+
+	}).catch((error)=>{
+		console.log(error)
+			toast.error(error.response.data.message || 'something wrong went please try again')
```

Comment:
We need to use optional chaining here to avoid having runtime errors.

Reply:
done